File: Miscellaneous/Golang/217. Contains Duplicate.go
package main

import "fmt"

func main() {

	arr := []int{1, 2, 3, 1}
	result := containsDuplicate(arr)
	fmt.Println(result)
}

func containsDuplicate(arr []int) bool {
	dict := make(map[int]bool)

	for i := 0; i < len(arr); i++ {
		ok := dict[arr[i]]
		if ok {
			return true
		}
		dict[arr[i]] = true
	}
	return false
}

File: Miscellaneous/Python/217. Contains Duplicate.py
class Solution:
    def containsDuplicate(self, arr):
        dict = {}
        for num in arr:
            if num not in dict.keys():
                dict[num] = 1
            else:
                return True
        return False

obj = Solution()
arr = [1, 2, 3, 1]
ans = obj.containsDuplicate(arr)
print(ans)